pkg/access_token/access_token_file.go

```diff
@@ -0,0 +1,69 @@
+package access_token
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type AccessTokenFile struct {
+	TokenFilePath string
+	Data          []byte
+}
+
+func NewAccessTokenFile(tokenFilePath string) (token *AccessTokenFile, err error) {
+	return &AccessTokenFile{
+		TokenFilePath: tokenFilePath,
+		Data:          nil,
+	}, nil
+}
+
+func (token AccessTokenFile) Read() (Data []byte, err error) {
+	if token.Data == nil {
+		return nil, fmt.Errorf("error reading access token, reason: data is empty")
+	}
+
+	return token.Data, nil
+}
+
+func (token *AccessTokenFile) Write(Data []byte) (err error) {
+	if Data == nil {
+		return fmt.Errorf("error writing access token, reason: data is empty")
+	}
+
+	token.Data = Data
+	// Write the data to file
+	// Create the directory if it doesn't exist
+	tokenDir := filepath.Dir(token.TokenFilePath)
+	if _, err := os.Stat(tokenDir); os.IsNotExist(err) {
+		err = os.MkdirAll(tokenDir, 755)
+		if err != nil {
+			// Do not specify the directory in the error message for security reasons
+			return fmt.Errorf("error writing access token, reason: failed to create directory")
```

Comment:
There's a few places where the error returned is an interpretation of the original error. I think that might have the effect of making debugging more difficult. I think it might warranted to return an interpreted error in certain circumstances. I might be wrong but I'm not immediately convinced that file paths and folder paths warrant it.

Reply:
not sure i got your comment here. what's the action item?

Reply:
What is the security argument for obscuring the original error with folder path and file name ?

I was thinking we should send the interpreted error concatenated with the original error. The errors coming from deeper in the stack provide meaningful information for debugging.

Reply:
@shaharglazner can you please take a look and see if we have some security concern here?

Reply:
So, there are two security concerns here. I agree these aren't fundamental security concerns, but it worth discussing it.

**The path to the token is somehow sensitive** 
OWASP #8.1 - 
```
Verify that the application does not output error messages or stack traces containing sensitive data that could assist an attacker, including session-id, software/framework versions, and personal information
```
So, the question here is if the path to the token can help the attacker to get it, or it doesn't matter because the attacker would find it anyway. I think it is sensitive in some way, therefore I think its worth to consider to specify the path only in DEBUG or something similar (or to add the ability to remove it from logs)


**The content of the path is a user-input and therefore could be used to manipulate the logs/the system that reads them**
OWASP #8.5 and #8.8 - 
```
8.5
Verify that all events that include untrusted data will not execute as code in the intended log viewing software.
```
```
8.8
Verify that all non-printable symbols and field separators are properly encoded in log entries, to prevent log injection.
```
So if we are going to log the path, we need to validate/sanitize if before just printing it.

WDYT? @orenbm @doodlesbykumbi

Reply:
i think that the value we can get from printing this is not high enough to take the risk or even further investigate it. I'll wait for @doodlesbykumbi 's thought on this before we proceed.

Reply:
Fair assessment @orenbm. Thanks for giving this some consideration.